ledger/nomos-ledger/src/utxotree.rs

```diff
@@ -0,0 +1,75 @@
+use nomos_core::mantle::{
+    merkle::{self, PathNode},
+    Note, NoteId, Utxo,
+};
+// up to 2^14 commitments
+const MAX_UTXOS: usize = 1 << 14;
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+pub struct UtxoTree {
+    utxos: rpds::RedBlackTreeMapSync<NoteId, Note>,
+}
+
+// TODO: change to sparse computation
+fn note_id_leaves(utxos: &rpds::RedBlackTreeMapSync<NoteId, Note>) -> [[u8; 32]; MAX_UTXOS] {
```

Comment:
Nitpick: `[[u8; 32]; MAX_UTXOS]` Hashes? I think at least a type alias would make it better to read? `MerkleLeaves`?